include/deal.II/fe/fe_base.h

```diff
@@ -240,6 +240,23 @@ class FiniteElementData
    */
   const unsigned int dofs_per_hex;
 
+  /**
+   * Number of non-zero degrees of freedom (DoFs) on the cell that,
+   * however, can not be associated with the specific geometric object (such as
+   * face or edge). In these cases, DoFs are typically globally distributed over
+   * (patches of) the Triangulation. Typical examples are:
+   *  - <a
+   * href="https://www.sciencedirect.com/science/article/pii/S0045782504005171">Isogeometric
+   * Analysis</a>
+   *  - <a
+   * href="https://onlinelibrary.wiley.com/doi/abs/10.1002/%28SICI%291097-0207%2819990910%2946%3A1%3C131%3A%3AAID-NME726%3E3.0.CO%3B2-J">Extended
+   * finite element method</a>
+   *  - <a
+   * href="http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.21.4515&rep=rep1&type=pdf">Catmull-Clark's
+   * subdivision surfaces</a>
+   */
+  const unsigned int non_local_dofs_per_cell;
```

Comment:
May I ask you to make this private and introduce a method `n_non_local_dofs_per_cell()`. 

I know this is not how we do it right now, however, my goal for the next few weeks is that we internally only use the getter functions (see also #10674). The motivation is that if we ever want to support wedges/pyramids we will need to be able to querry number of vertices/lines of i-th face.

Reply:
Done.